File: 225. Implement Stack using Queues/solution.js
/**
 * Initialize your data structure here.
 */
var MyStack = function() {
  this.list = [];
};

/**
* Push element x onto stack. 
* @param {number} x
* @return {void}
*/
MyStack.prototype.push = function(x) {
  this.list.unshift(x);
};

/**
* Removes the element on top of the stack and returns that element.
* @return {number}
*/
MyStack.prototype.pop = function() {
  return this.list.shift();
};

/**
* Get the top element.
* @return {number}
*/
MyStack.prototype.top = function() {
  return this.list[0];
};

/**
* Returns whether the stack is empty.
* @return {boolean}
*/
MyStack.prototype.empty = function() {
  return this.list.length === 0;
};

/** 
* Your MyStack object will be instantiated and called as such:
* var obj = Object.create(MyStack).createNew()
* obj.push(x)
* var param_2 = obj.pop()
* var param_3 = obj.top()
* var param_4 = obj.empty()
*/

File: 225. Implement Stack using Queues/solution.go
type MyStack struct {
	Queue    []int
	Height   int
	Capacity int
}

/** Initialize your data structure here. */
func Constructor() MyStack {
	return MyStack{Queue: []int{}, Height: 0, Capacity: 0}
}

/** Push element x onto stack. */
func (obj *MyStack) Push(x int) {
	if obj.Height == obj.Capacity {
		obj.Queue = append(obj.Queue, x)
		obj.Height++
		obj.Capacity++
	} else {
		obj.Queue[obj.Height] = x
		obj.Height++
	}
}

/** Removes the element on top of the stack and returns that element. */
func (obj *MyStack) Pop() int {
	if obj.Height > 0 {
		obj.Height--
		return obj.Queue[obj.Height]
	} else {
		panic("Stack is empty!")
	}
}

/** Get the top element. */
func (obj *MyStack) Top() int {
	if obj.Height > 0 {
		return obj.Queue[obj.Height-1]
	} else {
		panic("Stack is empty!")
	}
}

/** Returns whether the stack is empty. */
func (obj *MyStack) Empty() bool {
	return obj.Height == 0
}

/**
 * Your MyStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.Empty();
 */